### PR TITLE
feat(composer): composer.lock support

### LINKED ISSUE #2098: Composer lock file support

### PR DESCRIPTION
**Pull request by rarkins**
Adds logic to support updating the accompanying `composer.lock` file whenever `composer.json` is updated.

Closes #2098

### CODE REVIEW

File: lib/config/definitions.js [hunk 1]
```diff
@@ -1013,7 +1013,7 @@ const options = [
     stage: 'package',
     type: 'json',
     default: {
-      enabled: false,
+      enabled: true,
```
Comment (rarkins) [inline lib/config/definitions.js]: I might not enable it by default just yet though

**Comment by swashata** [comment]
Hi,

Do you think this can be merged as is? I've deployed the latest of renovateapp on my server and the changes made to composer.json seems to work fine.

**Comment by rarkins** [comment]
Yes, I just need to fix the test coverage first

**Comment by swashata** [comment]
maybe I've found a new issue #2281

**Comment by evilebottnawi** [comment]
@rarkins great work, thanks!

**Pull request merged**